frontend/src/views/OfferView.vue

```diff
@@ -1,60 +1,123 @@
 import Button from "primevue/button";
 import Calendar from "primevue/calendar";
 import Card from "primevue/card";
 import Dropdown from "primevue/dropdown";
 import FloatLabel from "primevue/floatlabel";
+import SplitButton from "primevue/splitbutton";
 import Textarea from "primevue/textarea";
-import { onMounted, ref } from "vue";
-import { useRoute } from "vue-router";
+import { useToast } from "primevue/usetoast";
+import { onMounted, ref, watch } from "vue";
+import { useRoute, useRouter } from "vue-router";
 
-import { getOrder } from "@/backendClient";
+import { createOffer, getOrder } from "@/backendClient";
 import OfferItem from "@/components/orders/OfferItem.vue";
 import { OfferStatus } from "@/global/types/appTypes";
 import { ArticleKind } from "@/global/types/appTypes";
-import type { OfferCreate } from "@/global/types/dataEditTypes";
+import type { OfferCreate, OfferItemCreate } from "@/global/types/dataEditTypes";
 import type { Offer, Order } from "@/global/types/entities";
 import { ROUTES } from "@/router";
 
 let itemCount = 1;
 const route = useRoute();
+const router = useRouter();
+
 const offersType = Object.values(OfferStatus);
 let orderInfo = ref<Order | undefined>();
-let offerDate = ref();
-let validityDate = ref();
+let offerDate = ref<Date>();
+let validityDate = ref<Date>();
 
 let offerInfo = ref<OfferCreate | Offer>({
   order_id: "",
-  order: orderInfo.value,
   status: OfferStatus.initial,
   description: "",
   offered_at: "",
   offer_valid_until: "",
   items: [],
 });
 
-let offerItemsArray = ref<any>([]);
+let offerItemsArray = ref<
+  {
+    id: number;
+    type: ArticleKind;
+  }[]
+>([]);
+
+/**
+ * Store information about sub-items in the object so we can easily update them.
+ */
+const itemsMap = ref<Record<string, OfferItemCreate>>({});
 
 function addOfferItem(type: ArticleKind) {
   offerItemsArray.value.push({ id: itemCount++, type: type });
 }
 
+const toast = useToast();
+const items = [
+  {
+    label: "Add note",
+    command: () => {
+      addOfferItem(ArticleKind.heading);
+      toast.add({ severity: "success", detail: "Note added", life: 3000 });
+    },
+  },
+  {
+    label: "Add position",
+    command: () => {
+      addOfferItem(ArticleKind.item),
+        toast.add({ severity: "success", detail: "Position added", life: 3000 });
+    },
+  },
+];
+
+async function onSaveOffer() {
+  const items = Object.values(itemsMap.value);
+
+  /**
+   * Add items from the itemsMap
+   */
+  await createOffer({
+    ...offerInfo.value,
+    items: items,
+  });
+
+  router.push(`${ROUTES.ORDER.path}/${route.params.order_id}/edit`);
+}
+
+watch(offerDate, () => {
+  if (offerDate.value) {
+    offerInfo.value.offered_at = format(offerDate.value, "yyyy-MM-dd");
+  } else {
+    offerInfo.value.offered_at = "";
+  }
+});
+
+watch(validityDate, () => {
+  if (validityDate.value) {
+    offerInfo.value.offer_valid_until = format(validityDate.value, "yyyy-MM-dd");
```

Comment:
Is format the same as "formatIsoDateString" here?

Reply:
we have another format and I added helper function for the formatting